File: app/src/main/java/com/example/marvelworld/model/MarvelApiData.java
package com.example.marvelworld.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class MarvelApiData {

        @SerializedName("offset")
        @Expose
        private Integer offset;
        @SerializedName("limit")
        @Expose
        private Integer limit;
        @SerializedName("total")
        @Expose
        private Integer total;
        @SerializedName("count")
        @Expose
        private Integer count;
        @SerializedName("results")
        @Expose
        private List<MarvelCharacter> marvelCharacters = null;

        public Integer getOffset() {
            return offset;
        }

        public void setOffset(Integer offset) {
            this.offset = offset;
        }

        public Integer getLimit() {
            return limit;
        }

        public void setLimit(Integer limit) {
            this.limit = limit;
        }

        public Integer getTotal() {
            return total;
        }

        public void setTotal(Integer total) {
            this.total = total;
        }

        public Integer getCount() {
            return count;
        }

        public void setCount(Integer count) {
            this.count = count;
        }

        public List<MarvelCharacter> getMarvelCharacters() {
            return marvelCharacters;
        }

        public void setResults(List<MarvelCharacter> marvelCharacters) {
            this.marvelCharacters = marvelCharacters;
        }
}

File: app/src/main/java/com/example/marvelworld/model/MarvelApiData.kt
package com.example.marvelworld.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class MarvelApiData {
    @SerializedName("offset")
    @Expose
    var offset: Int? = null
    @SerializedName("limit")
    @Expose
    var limit: Int? = null
    @SerializedName("total")
    @Expose
    var total: Int? = null
    @SerializedName("count")
    @Expose
    var count: Int? = null
    @SerializedName("results")
    @Expose
    var marvelCharacters: List<MarvelCharacter>? = null
        private set

    fun setResults(marvelCharacters: List<MarvelCharacter>?) {
        this.marvelCharacters = marvelCharacters
    }
}